vllm/config.py

```diff
@@ -927,6 +927,10 @@ def maybe_create_spec_config(
                     "Speculative decoding with mlp_speculator models does not "
                     "yet support distributed inferencing (TP > 1).")
 
+            if draft_model_config.hf_config.model_type == "medusa":
+                draft_model_config.hf_config.set_num_lookahead_tokens(
+                    num_speculative_tokens)
```

Comment:
@abhigoyal1997 I just merged https://github.com/vllm-project/vllm/pull/5876 which fixes the equivalent thing for MLPSpeculator.

Was thinking you could adjust the medusa config to expose `num_lookahead_tokens` as a attribute/property instead of a custom setter (could use python property annotation).. then hopefully the code to set it here can be common and doesn't need to check the explicit model type. WDYT?

Reply:
modified.